src/devices/Tlc1543/Tlc1543.cs

```diff
@@ -0,0 +1,221 @@
+        /// <remarks>
+        /// <br>Refer to table 2 and 3 in TLC1543 documentation for more information</br>
+        /// </remarks>
+        /// </summary>
+        public Channel ChargeChannel
+        {
+            get => _chargeChannel;
+            set
+            {
+                if (IsValidChannel((int)value))
+                {
+                    _chargeChannel = value;
+                }
+                else
+                {
+                    throw new ArgumentOutOfRangeException("ChargeChannel", "Out of range. \nMust be non-negative and less than 14. \nRefer to table 2 and 3 in TLC1543 documentation for more information");
+                }
+            }
+        }
+
+        #endregion
+
+        #region Constructors
+
+        /// <summary>
+        /// Constructor for Tlc1543
+        /// </summary>
+        /// <param name="address">Address pin</param>
+        /// <param name="chipSelect">Chip Select pin</param>
+        /// <param name="dataOut">Data Out pin</param>
+        /// <param name="inputOutputClock">I/O Clock pin</param>
+        /// <param name="endOfConversion">End of Conversion pin</param>
+        /// <param name="controller">The GPIO controller for defined external pins. If not specified, the default controller will be used.</param>
+        /// <param name="pinNumberingScheme">Pin Numbering Scheme</param>
+        /// <param name="shouldDispose">True to dispose the Gpio Controller</param>
+        public Tlc1543(int address, int chipSelect, int dataOut, int inputOutputClock, int endOfConversion = -1, GpioController controller = null, PinNumberingScheme pinNumberingScheme = PinNumberingScheme.Logical, bool shouldDispose = true)
+        {
+            _shouldDispose = controller == null || shouldDispose;
+            _controller = controller ?? new GpioController(pinNumberingScheme);
+            _address = address;
+            _chipSelect = chipSelect;
+            _dataOut = dataOut;
+            _endOfConversion = endOfConversion;
+            _inputOutputClock = inputOutputClock;
+
+            _controller.OpenPin(_address, PinMode.Output);
+            _controller.OpenPin(_chipSelect, PinMode.Output);
+            _controller.OpenPin(_dataOut, PinMode.InputPullUp);
+            _controller.OpenPin(_inputOutputClock, PinMode.Output);
+            if (_endOfConversion != -1)
+            {
+                _controller.OpenPin(_endOfConversion, PinMode.InputPullUp);
+            }
+        }
+
+        #endregion
+
+        #region Methods
+
+        /// <summary>
+        /// Check if channelNumber is valid
+        /// </summary>
+        /// <param name="channelNumber">Channel Address</param>
+        /// <returns></returns>
+        protected static bool IsValidChannel(int channelNumber)
+        {
+            if (channelNumber < 0 || channelNumber > 13)
+            {
+                return false;
+            }
+
+            return true;
+        }

```

Comment:
```suggestion
        protected static bool IsValidChannel(int channelNumber) => channelNumber >= 0 && channelNumber <= 13;
```

although the exception message in the constructor said "positive" which suggests (1-13) - please correct one or another